chain/chain/src/types.rs

```diff
@@ -79,25 +79,24 @@ pub struct AcceptedBlock {
 }
 
 #[derive(Debug)]
-pub struct ApplySplitStateResult {
+pub struct ApplyResultForResharding {
     pub shard_uid: ShardUId,
     pub trie_changes: WrappedTrieChanges,
     pub new_root: StateRoot,
 }
 
-// This struct captures two cases
-// when apply transactions, split states may or may not be ready
-// if it's ready, apply transactions also apply updates to split states and this enum will be
-//    ApplySplitStateResults
-// otherwise, it simply returns the state changes needed to be applied to split states
+// ReshardingResults contains the results of applying depending on whether
+// resharding is finished.
+// If resharding is finished the results should be applied immediately.
+// If resharding is not finished the results should be stored and applied later.
 #[derive(Debug)]
-pub enum ApplySplitStateResultOrStateChanges {
-    /// Immediately apply the split state result.
+pub enum ReshardingResults {
+    /// Immediately apply the resharding result.
     /// Happens during IsCaughtUp and CatchingUp
-    ApplySplitStateResults(Vec<ApplySplitStateResult>),
-    /// Store the split state results so that they can be applied later.
+    ApplyReshardingResults(Vec<ApplyResultForResharding>),
+    /// Store the resharding results so that they can be applied later.
     /// Happens during NotCaughtUp.
-    StateChangesForSplitStates(StateChangesForSplitStates),
+    StoreReshardingResults(StateChangesForResharding),
```

Comment:
I guess the name `ReshardingResult` here is fine but we might want to change the two enums under it. `StoreReshardingResults` can be tweaked to `StoredReshardingResult` indicating it's saved in DB?

Reply:
Similarly `AppliedReshardingResult` and `AppliedReshardingResults` -> to be applied. Couldn't really come up with anything better.

Reply:
I think it's only when the ReshardingResults is processed in `process_resharding_results` that it goes into past tense.

Reply:
Yeah, let's just keep as is... All good